x/axelarnet/types/msg_link.go

```diff
@@ -33,14 +34,16 @@ func (m LinkRequest) ValidateBasic() error {
 		return sdkerrors.Wrap(sdkerrors.ErrInvalidAddress, sdkerrors.Wrap(err, "sender").Error())
 	}
 
-	if m.RecipientChain == "" {
-		return fmt.Errorf("missing recipient chain")
+	if err := utils.ValidateString(m.RecipientChain, false); err != nil {
```

Comment:
We need to do this wherever we accept a string as input. Might be worth having a type alias for safety and readability if it seems helpful. The validation function for this type can also check that no underscore/separator was present.

Reply:
My intention is to do this for all places where we accept a string as input, hence the draft PR. But I wanted to first give everybody a chance to look at it to decide if this is the approach we want.

I am not so sure about the type alias because we probably should have the option of accepting empty strings, and if we used an alias we would either have to require an boolean arg in the Validate function, or to create a an actual new struct. I do agree that we should extent the current function to disallow whatever separator we use in the keepers (maybe even another argument with a list of forbidden characters).

Reply:
Makes sense to test this on one case.

Regarding empty strings, I think the alternative is to have two validate methods. Normal `Validate` shouldn't allow empty strings (as this is by far the usual case). And another method like `ValidateAllowEmpty` can explicitly allow it. Passing along a boolean arg everywhere is more error-prone, so an alternative like this would be safer imo.

Reply:
or maybe two different alias "NormalizedString" and "NormalizedNonEmptyString"